modules/javafx.graphics/src/main/java/com/sun/javafx/font/FallbackResource.java

```diff
@@ -196,27 +216,48 @@ public int getSlotForFont(String fontName) {
                 i++;
             }
         }
+        return -1;
+    }
 
-        if (i >= 0x7E) {
-            /* There are 8bits (0xFF) reserved in a glyph code to store the slot
-             * number. The first bit cannot be set to avoid negative values
-             * (leaving 0x7F). The extra -1 (leaving 0x7E) is to account for
-             * the primary font resource in PrismCompositeFontResource.
-             */
-            if (PrismFontFactory.debugFonts) {
-                System.err.println("\tToo many font fallbacks!");
-            }
-            return -1;
+    @Override
+    public int getSlotForFont(String fontName) {
+      int slot = getSlotForFontNoCreate(fontName);
+        if (slot >= 0) {
+            return slot;
         }
+
         PrismFontFactory factory = PrismFontFactory.getFontFactory();
         FontResource fr = factory.getFontResource(fontName, null, false);
+
         if (fr == null) {
             if (PrismFontFactory.debugFonts) {
                 System.err.println("\t Font name not supported \"" + fontName + "\".");
             }
             return -1;
         }
+        slot = getSlotForFontNoCreate(fr.getFullName());
+        if (slot >= 0) {
+            return slot;
+        }
+
+        /* Add the font to the list of native fallbacks */
+        return addNativeFallback(fr);
+    }
+
 
+    private int addNativeFallback(FontResource fr) {
+        int ns = getNumSlots();
+        if (ns >= 0x7E) {
+            /* There are 8bits (0xFF) reserved in a glyph code to store the slot
+             * number. The first bit cannot be set to avoid negative values
+             * (leaving 0x7F). The extra -1 (leaving 0x7E) is to account for
+             * the primary font resource in PrismCompositeFontResource.
+             */
+            if (PrismFontFactory.debugFonts) {
+                System.err.println("\tToo many font fallbacks!");
```

Comment:
would it be better to use logging?

Reply:
I can't change just the new cases. And changing hundreds of other cases is out of scope.